### PR TITLE
Add recipe for cfitsio

### PR DESCRIPTION
**Pull request by saimn**

### CODE REVIEW

File: requirements.yml [hunk 1]
```diff
@@ -66,3 +66,7 @@
   numpy_build_restrictions: '>=1.9*,<1.11'
   excluded_platforms:
       - 'win-64'
+
+- name: cfitsio
+  version: '3.390'
+  md5: e92dd2a4282a1c50d46167041a29fc67
```
Comment (saimn) [inline requirements.yml]: Not sure if the md5 can be specified here ?

**Comment by saimn** [comment]
The build fails with a Jinja error:

```
AttributeError: <module 'conda_build.jinja_context' from '/home/travis/miniconda/lib/python2.7/site-packages/conda_build/jinja_context.pyc'> does not have the attribute 'select_lines'
```

I didn't have this issue when running travis on my fork of conda-channel-astropy

**Comment by mwcraig** [comment]
I should be able to fix this Sunday or Monday, am going offline for a couple days. Think the fix is in the travis/appveyor setup.

What version of conda-build are you using?

**Comment by saimn** [comment]
You mean when using conda-channel-astropy ? It seems to use only the patched 1.18.1 version, whereas here you use conda-build=1.20.2 which doesn't have this `select_lines` function. Locally I have conda-build-1.21.5 which do have `select_lines` . Should I try to upgrade it in the travis config ?

**Comment by mwcraig** [comment]
@saimn -- take a look here too, @pkgw has a conda recipe for cfitsio: https://github.com/pkgw/conda-recipes/

**Comment by mwcraig** [comment]
The update of conda-build should help. You could also update the patched conda-build version to 1.21.1, and in a moment I'll get the patched version update to 1.21.5.

Next week at scipy I'll try make sure this patch gets merged into conda-build.

**Comment by mwcraig** [comment]
Fail is related to https://github.com/conda/conda/issues/3004

**Comment by mwcraig** [comment]
Closing and reopening this to trigger a fresh build....

File: .travis.yml [hunk 2]
```diff
@@ -61,6 +61,6 @@ script:
     # the latest known good version. It contains a fix for an issue in which
     # Windows is unwilling to empty the trash, causing a problem.
     - conda remove --force conda-build
-    - conda install conda-build=1.20.2
+    - conda install conda-build-1.21.5
```
Comment (mwcraig) [inline .travis.yml]: you have **-**`1.21.5` not **=**`1.21.5`

Comment (saimn) [inline .travis.yml]: Ohh, sorry ! :/

**Comment by saimn** [comment]
> @pkgw has a conda recipe for cfitsio: https://github.com/pkgw/conda-recipes/

I saw it when searching for cfitsio in anaconda.org, but it is an old version so I didn't know if it was maintained, and I didn't know how to find the recipe.
In my recipe I also added the `--enable-reentrant`option which is required for multi-threaded env or use with OpenMP, otherwise the recipes are very similar.

**Comment by saimn** [comment]
Build is ok this time, it fails when trying to upload the package, which is expected I guess.

**Comment by saimn** [comment]
Except Appveyor is building on py27, 34 and 35. Any way to avoid this ?

**Comment by saimn** [comment]
Hi @mwcraig , any news on this ?

**Comment by mwcraig** [comment]
@saimn -- thanks for the ping; I've been iterating on conda-channel-astropy. Let me move those changes over here, then restart the builds here.

What versions of python should be excluded on Windows?

**Comment by saimn** [comment]
Thanks for the update. Shall I port my branches for cfitsio and fitsio to conda-channel-astropy ? 
The cfitsio build is independent from Python, so it should be built only once.

**Comment by mwcraig** [comment]
No, let's keep it here. I think the desire is to keep `conda-channel-astropy` as affiliated packages and their dependencies only, with openastronomy as a superset of that.

Just opened #19, which should allow this to succeed after a rebase (or just back out the change in `.travis.yml` in your PR).

By later today, hopefully, we'll be able to use the official conda-build.

**Comment by mwcraig** [comment]
@saimn -- could you please rebase on master when you get a chance? You can drop your changes in `.travis.yml` too.

**Comment by saimn** [comment]
> No, let's keep it here. I think the desire is to keep conda-channel-astropy as affiliated packages and their dependencies only, with openastronomy as a superset of that.

Ok, sorry I misunderstood your previous comment ;)
I rebased the branch.

**Comment by mwcraig** [comment]
Green lights, so merging! Give this about 15 minutes, then rebase the `fitsio` PR...

**Pull request merged**